### PR TITLE
Fix reading scope attributes for trace JSON, remove duplicate code

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/5312 

Signed-off-by: Bogdan <bogdandrutu@gmail.com>

### CODE REVIEW

File: pdata/internal/json/scope.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	jsoniter "github.com/json-iterator/go"
+
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+)
+
+func ReadScope(iter *jsoniter.Iterator, scope *otlpcommon.InstrumentationScope) {
+	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+		switch f {
+		case "name":
```
Comment (TylerHelmuth) [inline pdata/internal/json/scope.go:26]: Any chance there are constants from the spec/proto we can use to represent these cases?

File: pdata/internal/json/scope_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import (
+	"testing"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/stretchr/testify/assert"
+
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+)
+
+func TestReadScope(t *testing.T) {
```
Comment (TylerHelmuth) [inline pdata/internal/json/scope_test.go:26]: Can you add a test case that covers the situation where a field is skipped?

File: pdata/internal/json/scope.go (hunk 1 above)
Comment (bogdandrutu) [reply pdata/internal/json/scope.go:26]: They are tags, but that needs reflection package and is expensive. Not anything else that I know, but probably we can do a proto plugin to extract these values.

Comment (TylerHelmuth) [reply pdata/internal/json/scope.go:26]: Looking through pdata seems like we use this strategy a lot, so changing,  if possible, would probably be better for a later PR.

File: pdata/internal/json/scope_test.go (hunk 2 above)
Comment (bogdandrutu) [reply pdata/internal/json/scope_test.go:26]: Done.

File: pdata/internal/json/scope.go (hunk 1 above)
Comment (bogdandrutu) [reply pdata/internal/json/scope.go:26]: Definitely will not do this in the current PR, also needs a lot of changes :)

File: pdata/internal/json/scope.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+package json // import "go.opentelemetry.io/collector/pdata/internal/json"
+
+import (
+	jsoniter "github.com/json-iterator/go"
+
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+)
+
+func ReadScope(iter *jsoniter.Iterator, scope *otlpcommon.InstrumentationScope) {
```
Comment (tigrannajaryan) [inline pdata/internal/json/scope.go:23]: Do we need to also call ReadScope for logs? I don't see it in this PR.

**Review by tigrannajaryan — APPROVED** [review]

File: pdata/internal/json/scope.go (hunk 3 above)
Comment (bogdandrutu) [reply pdata/internal/json/scope.go:23]: Logs was not yet converted to the new `jsoniter`, still uses `jsonpb`.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**